pkg/chunk/purger/purger.go

```diff
@@ -24,6 +25,42 @@ import (
 
 const millisecondPerDay = int64(24 * time.Hour / time.Millisecond)
 
+type purgerMetrics struct {
+	deleteRequestsProcessedTotal      *prometheus.CounterVec
+	deleteRequestsChunksSelectedTotal *prometheus.CounterVec
+	deleteRequestsProcessingFailures  *prometheus.GaugeVec
+}
+
+func newPurgerMetrics(r prometheus.Registerer) *purgerMetrics {
+	m := purgerMetrics{}
+
+	m.deleteRequestsProcessedTotal = prometheus.NewCounterVec(prometheus.CounterOpts{
```

Comment:
Instead of calling `r.MustRegister()` below, could you create the metrics with `promauto.With(r).NewCounterVec()` (and same for `NewGaugeVec()`)? It's the pattern we're heading to.